### DISCUSSION TITLE
How to get the check_run_id of GitHub Actions?

### DISCUSSION THREAD

**Discussion by ElangoRajendran** [cake-build/cake#3334] [Q&A]
I have run my build.cake file tasks in PowerShell script on the windows machine. Do I need to get the check_run_id of GitHub Actions in the workflows .yml file?

**i.e. https://github.com/Siddharth/my-project/pull/15/checks?check_run_id=2508655272**

I need this id(2508655272) value like this in the workflows .yml file. I have attached my workflow file.

Is there any way to get check_run_id?
Could you suggest a solution for how to achieve this in my workflow .yml file?

![image](https://user-images.githubusercontent.com/59197465/117138734-0f338b00-adc9-11eb-8aca-50e86cf20e46.png)

**Comment by nils-a** [comment]
@ElangoRajendran 
https://github.community/t/getting-the-check-run-id-in-github-actions/126243/2 suggest that id is requestable via REST services:

> based on the ${{ github.run_id}}, use rest api to get the check_suite_url. https://developer.github.com/v3/actions/workflow-runs/#get-a-workflow-run
> Based on the check_suite_url, get the check_run_id via rest api: https://developer.github.com/v3/checks/runs/#list-check-runs-in-a-check-suite

So it seems there is no easy way to get to that information.

You could try the way https://github.community/t/getting-the-check-run-id-in-github-actions/126243/2 suggests. I would use `Cake.Curl` to access the API and then `Cake.JSON` to parse the returned information.
